### PR TITLE
Add `HealthPlug` addon

### PR DESCRIPTION
**Pull request by andyduong1920**
https://github.com/nimblehq/elixir-templates/issues/155

## What happened

Add the HealthPlug

## Insight

1/ copy the HealthPlug and HealPlugTest
2/ Add the health_path into the configs
3/ Add the forwarder in the router

## Proof Of Work

1/ The web part

image

2/ The test part

image

3/ The config part

image

4/ The wiki part

image

### CODE REVIEW

**Review by topnimble — APPROVED** [review]

File: priv/templates/nimble_template/.github/wiki/_Sidebar.md [hunk 1]
```diff
@@ -2,3 +2,8 @@
 
 - [[Home]]
 - [[Getting Started]]
+
+## Infrastructure
+
+- [[Application Status]]
+- [[Configuring Variables|Variable Configuration]]
```
Comment (rosle) [inline priv/templates/nimble_template/.github/wiki/_Sidebar.md:9]: Should we consider using the same name? 🤔 💭  either updating the title or the filename.

File: lib/nimble_template/addons/variants/phoenix/health_plug.ex [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  end
+
+  defp edit_test_helper(project) do
+    Generator.replace_content(
+      "test/test_helper.exs",
+      """
+      ExUnit.start()
+      """,
+      """
+      Mimic.copy(Ecto.Adapters.SQL)
```
Comment (rosle) [inline lib/nimble_template/addons/variants/phoenix/health_plug.ex:109]: Does this depend on Mimic addon? 🤔  Or we're always applying the mimic addon, right? That should be ok? 💭

File: priv/templates/nimble_template/test/otp_app_web/plugs/health_plug_test.exs.eex [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  alias <%= web_module %>.HealthPlug
+
+  describe "init/1" do
+    test "returns given options" do
+      assert HealthPlug.init([]) == []
+    end
+  end
+
+  describe "call/2" do
+    test "returns 200 status given the liveness path" do
```
Comment (rosle) [inline priv/templates/nimble_template/test/otp_app_web/plugs/health_plug_test.exs.eex:15]: Should we update the test description to align with the new convention 💡 
```suggestion
    test "given the liveness path, returns 200 status" do
```

File: test/nimble_template/addons/variants/health_plug_test.exs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      in_test_project(test_project_path, fn ->
+        PhoenixAddons.HealthPlug.apply(project)
+
+        assert_file("test/nimble_template_web/plugs/health_plug_test.exs")
+        assert_file("test/nimble_template_web/requests/_health/liveness_request_test.exs")
+        assert_file("test/nimble_template_web/requests/_health/rediness_request_test.exs")
+      end)
+    end
+
+    test "adds health_path configuration in config", %{
```
Comment (rosle) [inline test/nimble_template/addons/variants/health_plug_test.exs:31]: Sometimes we're referring `health_path` or `health path`. Should we update it to be consistent 💡 
```suggestion
    test "adds health path configuration in config", %{
```

File: lib/nimble_template/addons/variants/phoenix/health_plug.ex (hunk 2 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/health_plug.ex:109]: Yes Ros, we need the Mimic in the `requests test` and we always pass the Mimic addon :)

File: test/nimble_template/addons/variants/health_plug_test.exs (hunk 4 above)
Comment (andyduong1920) [reply test/nimble_template/addons/variants/health_plug_test.exs]: adjusted it in https://github.com/nimblehq/elixir-templates/pull/159/commits/06a24ca05d89961279d29d39e0542e117dafe7c4 @rosle

File: priv/templates/nimble_template/test/otp_app_web/plugs/health_plug_test.exs.eex (hunk 3 above)
Comment (andyduong1920) [reply priv/templates/nimble_template/test/otp_app_web/plugs/health_plug_test.exs.eex]: adjusted it in https://github.com/nimblehq/elixir-templates/pull/159/commits/06a24ca05d89961279d29d39e0542e117dafe7c4 @rosle

File: priv/templates/nimble_template/.github/wiki/_Sidebar.md (hunk 1 above)
Comment (andyduong1920) [reply priv/templates/nimble_template/.github/wiki/_Sidebar.md]: changed it in https://github.com/nimblehq/elixir-templates/pull/159/commits/06a24ca05d89961279d29d39e0542e117dafe7c4 @rosle

File: lib/nimble_template/addons/mimic.ex [hunk 5]
```diff
@@ -37,4 +38,45 @@ defmodule NimbleTemplate.Addons.Mimic do
...
+
+      """
+    )
+
+    project
+  end
+
+  defp edit_case(project) do
+    project
+  end
```
Comment (bterone) [inline lib/nimble_template/addons/mimic.ex:81]: Shall we put this in a single-line?

**Review by bterone — APPROVED** [review]

**Review by rosle — APPROVED** [review]

File: lib/nimble_template/addons/mimic.ex (hunk 5 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/mimic.ex:81]: Hi @bterone we don't use much single-line in the current template convention 😂 so I will keep it as its :D

**Pull request merged**